src/hnsw.c

```diff
@@ -94,6 +96,29 @@ hnswcostestimate(PlannerInfo *root, IndexPath *path, double loop_count,
 		return;
 	}
 
+	/* Get the selectivity of non-index conditions */
+	foreach(lc, path->indexinfo->indrestrictinfo)
+	{
+		RestrictInfo *rinfo = lfirst(lc);
+
+		if (rinfo->norm_selec >= 0 && rinfo->norm_selec <= 1 && rinfo->norm_selec != (Selectivity) DEFAULT_INEQ_SEL)
```

Comment:
The `DEFAULT_INEQ_SEL` check ensures the index is still used for `WHERE v <-> '[1,2,3]' < 1` (when there's an `ORDER BY` and `LIMIT`). This also means it's used for `WHERE v <-> '[1,2,3]' > 1`, which shouldn't really use an index (however, it currently does in existing releases).

Reply:
Is there a case where `rinfo->norm_selec` falls outside of `[0.0,1.0]`?

Reply:
From `pathnodes.h`:

```text
selectivity for "normal" (JOIN_INNER) semantics; -1 if not yet set; >1 means a redundant clause
```

I'm not sure if there are cases where it'll be out of that range when they get to this function, but seems better to be safe.